apps/checkout/src/providers/CountrySelectProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { PropsWithChildren, useState } from "react";
+import { createSafeContext } from "@/providers/createSafeContext";
+import { AddressFragment, CountryCode } from "@/graphql";
+
+const defaultCountryCode: CountryCode = "PL";
+
+interface CountrySelectProviderProps {
+  selectedCountryCode?: CountryCode;
+}
+
+export interface CountrySelectContextConsumerProps {
+  countryCode: CountryCode;
+  setCountryCode: (code?: CountryCode) => void;
+  setCountryCodeFromAddress: (address?: AddressFragment | null) => void;
+}
+
+export const [useCountrySelect, Provider] =
+  createSafeContext<CountrySelectContextConsumerProps>();
+
+export const CountrySelectProvider: React.FC<
+  PropsWithChildren<CountrySelectProviderProps>
+> = ({ children, selectedCountryCode }) => {
+  const [countryCode, handleSetCountryCode] = useState<CountryCode>(
```

Comment:
State `countryCode` and prop `setCountryCode` names are confusing. Maybe the prop should be named with prefix `handler`?

Reply:
I don't think I understand, what's confusing here?